lib/cdo/test_flakiness.rb

```diff
@@ -30,20 +28,29 @@ def self.calculate_test_flakiness
       )
       jobs += JSON.parse(response.body)
     end
+    return jobs
+  end
 
-    jobs_by_name = jobs.group_by {|job| job['name']}
-
-    name_to_flakiness = {}
+  def self.summarize_by_job
```

Comment:
Swap `each` for `map` to express this method as a single expression - just a little simpler.
```ruby
def self.summarize_by_job
  get_jobs.group_by {|job| job['name']}.map do |name, samples|
    {
      name: name,
      total: samples.count,
      failed: samples.count {|job| !job['passed']}
    }
  end
end
```

Reply:
Updated. Also, collapsed `get_jobs` and `summarize_by_job` into a single method. Although, I wonder if there is still benefit to having a separate `get_jobs` to see the real raw data. I was curious to see how the results are impacted if we tune the number of results from SauceLabs. Like how old are the results we are looking at, and is it worth considering failed tests that are too old. I was trying to avoid having to pass the `numRequests` and `perRequest` values around.